### ISSUE TITLE
Консольный режим

### ISSUE THREAD

**Issue by plotn** [petrovvlad/freeLib#52]
хотелось бы, чтобы программа умела работать в консольном режиме. 
Зачем это мне: хочу поднять opds на одноплатнике. Сейчас вроде задача решается, но приходится тащить кучу всего - qt, vnc и запускать из под этого всего, а хочется же systemctl enable ... и забыл.

Хватило бы, на самом деле, небольшого "api", навскидку:

create library [name] [path]
server opds [port]

Ну, остальное по мелочи.

Можно ли рассчитывать? Если бы мог, сам бы сделал, но я в С плоховат.
Если интересно - мой проект читалки - knownreader, приходите в гости на 4pda.

**Comment by petrovvlad** [comment]
Я уже думал об этом, но не был уверен, что это кому-то нужно. Сделаю.

**Comment by plotn** [comment]
Я успешно поднял софтинку на одноплатнике и она мне всю флибусту отдает с моего же локально подключенно по УСБ жесткого диска, при этом мой кноунридер спокойно читает ее, так что "это кому то нужно". Я просто не знал про вашу программу, поэтому только сейчас пришел ;)

**Comment by petrovvlad** [comment]
Добавил консольный режим. Пока только простой сервер. Для запуска сервера ключ --server.

**Comment by plotn** [comment]
проверил. Под vnc работает, под чистой консолью нет:

plotn@orangepi4:~/github/freeLib/build$ freelib --server
qt.qpa.xcb: could not connect to display
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

Аварийный останов

**Comment by petrovvlad** [comment]
Поправил. Проверяйте :)

**Comment by plotn** [comment]
А вы точно из девелопа в мастер влили? А то похоже что нет, но могу ошибаться

**Comment by petrovvlad** [comment]
теперь влил.

**Comment by plotn** [comment]
Да, стало лучше. Но не до конца. Пытаюсь получить список книг по автору - падает:

plotn@orangepi4:~/github/freeLib/build$ Open DB OK.  "/home/plotn/.local/share/freeLib/freeLib/freeLib.sqlite"
loadGenre  1 msec
loadSeria  121 msec
loadAuthor  706 msec
loadBooks  8699 msec
url: "/opds_1"
url: "/opds_1/authorsindex"
url: "/opds_1/authorsindex/Ж"
url: "/opds_1/authorsindex/Жа"
url: "/opds_1/authorsindex/Жав"
url: "/opds_1/author/154691"
QSqlQuery::value: not positioned on a valid record
url: "/opds_1/authorbooks/154691"
QPlatformPixmap: QGuiApplication required

**Comment by plotn** [comment]
Т е падает совсем, приходится перезапускать

**Comment by petrovvlad** [comment]
Исправил. Надеюсь в этот раз заработает.

**Comment by plotn** [comment]
да, работает! спасибо
можно ли рассчитывать на развитие консольного режима?

**Comment by petrovvlad** [comment]
Да, планирую. Пока добавил "--lib list" - выводит список библиотек.

**Comment by plotn** [comment]
В идеале надо конечно обеспечить возможность конфигурирования без запуска графики

**Comment by petrovvlad** [comment]
Настройки сохраняются в `~/.config/freeLib/freeLib.conf`, там их можно изменять без запуска графики. Работу с библиотеками (добавление, удаление, обновление) планирую добавить в консоль. Но это не скоро получится.

**Comment by petrovvlad** [comment]
Добавлены команды: _обновления, удаления, добавления_ библиотек.
